### PR TITLE
CassandraSinkCluster keyspace based routing - handle use statements

### PR DESCRIPTION
**Pull request by rukai**
This PR handles use statements in every case except one:
* operates on both batch operations and query operations. 
* messages are modified such that if a keyspace is not specified the `use` keyspace is used, this ensures a single source of truth for transforms.
* messages are modified at the codec level ensuring all layers of shotover observe the same keyspace.

The one case we arent handling is prepared statements.
use statements also affect prepared statements, but currently shotover does not support parsing prepared statements so I have just left that functionality as a TODO.

There is a performance concern: calling metadata is currently more expensive than it needs to be due to internally copying the entire message body and then dropping it.
But that can be addressed later on in a follow up PR.

Making this change caused the `.as_raw_bytes()` hack to start blowing up due to the message already being parsed, so I also changed all instances of it to just use `.metadata()` instead.

### CODE REVIEW

File: shotover-proxy/src/codec/cassandra.rs [hunk 1]
```diff
@@ -89,6 +110,70 @@ impl Decoder for CassandraCodec {
...
+            if let CassandraStatement::Use(keyspace) = query.as_ref() {
+                return Some(keyspace.clone());
+            }
+        }
+    }
+    None
+}
+
+fn set_default_keyspace(message: &mut Message, keyspace: &Identifier) {
+    // TODO: rewrite Operation::Prepared in the same way
```
Comment (conorbros) [inline shotover-proxy/src/codec/cassandra.rs:125]: Not sure what this means. What needs to be done for `Operation::Prepared`?

Comment (rukai) [reply shotover-proxy/src/codec/cassandra.rs:125]: The cassandra server will apply the current `use`d keyspace to Operation::Prepared messages that it receives.
To make that visible to transforms we should apply the same modifications that we apply to queries and batches to prepared statements as well

Comment (conorbros) [reply shotover-proxy/src/codec/cassandra.rs:125]: Yes I remember reading something interesting about prepared queries and keyspaces but I can't find it now. IIRC:

- it applies the used keyspace to `PREPARE` statements
- The keyspace set when the `PREPARE` statement was sent takes precedence over the current used keyspace (if it has changed)

Comment (rukai) [reply shotover-proxy/src/codec/cassandra.rs:125]: yep, that matches my understanding too

**Review by benbromhead — APPROVED** [review]

**Review by conorbros — APPROVED** [review]

**Pull request merged**